### PR TITLE
prod_cron_meta_nightly_pipeline getUMBFromVersion

### PR DESCRIPTION
**Pull request by Ginxo**
**Thank you for submitting this pull request**

caused by: https://github.com/kiegroup/kie-jenkins-scripts/pull/1118

Tested by https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/PROD/job/cron-meta-nightly-pipeline/160/console

<pre>
How to retest a PR or trigger a specific build:

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
</pre>

### CODE REVIEW

**Review by mareknovotny — APPROVED** [review]

File: job-dsls/jobs/prod/prod_cron_meta_nightly_pipeline.groovy [hunk 1]
```diff
@@ -94,7 +94,7 @@ pipeline{
             steps {
                 build job: 'kogito-tooling.nightly/0.13.0-prerelease', propagate: false, wait: true, parameters: [
                         [\$class: 'StringParameterValue', name: 'DEPLOYMENT_REPO_URL', value: 'https://bxms-qe.rhev-ci-vms.eng.rdu2.redhat.com:8443/nexus/service/local/repositories/scratch-release-rhba-7.12/content-compressed'],
```
Comment (winklerm) [inline job-dsls/jobs/prod/prod_cron_meta_nightly_pipeline.groovy:96]: We could derive the repository name in a similar way as UMB version, wdyt?

Comment (winklerm) [reply job-dsls/jobs/prod/prod_cron_meta_nightly_pipeline.groovy:96]: And maybe have a constant for the product branch (7.59.x), so there would be no need to update this part of the pipeline.

Comment (mareknovotny) [reply job-dsls/jobs/prod/prod_cron_meta_nightly_pipeline.groovy:96]: is this actually kogito-tooling needed here? we can remove/comment it out this step

Comment (mareknovotny) [reply job-dsls/jobs/prod/prod_cron_meta_nightly_pipeline.groovy:96]: and if we change and unify on string without dot it could be even less variable

**Review by mareknovotny — APPROVED** [review]
this looks ok

File: job-dsls/jobs/prod/prod_cron_meta_nightly_pipeline.groovy [hunk 2]
```diff
@@ -21,69 +36,69 @@ pipeline{
...
             steps {
-                build job: 'kogito.nightly/main', propagate: false, wait: true, parameters: [
-                        [\$class: 'StringParameterValue', name: 'RHBA_MAVEN_REPO_URL', value: 'http://bxms-qe.rhev-ci-vms.eng.rdu2.redhat.com:8081/nexus/content/repositories/rhba-main-nightly-with-upstream'],
-                        [\$class: 'StringParameterValue', name: 'RHBA_VERSION_PREFIX', value: "${RHBA_VERSION_PREFIX}"],
-                        [\$class: 'StringParameterValue', name: 'KOGITO_DEPLOYMENT_REPO_URL', value: 'https://bxms-qe.rhev-ci-vms.eng.rdu2.redhat.com:8443/nexus/service/local/repositories/scratch-release-kogito-main/content-compressed'],
-                        [\$class: 'StringParameterValue', name: 'UMB_VERSION', value: 'main'],
-                        [\$class: 'StringParameterValue', name: 'PRODUCT_VERSION', value: "${PRODUCT_VERSION}"],
-                        [\$class: 'StringParameterValue', name: 'OPTAPLANNER_PRODUCT_VERSION', value: "${PRODUCT_VERSION}"],
+                build job: 'kogito.nightly/${KOGITO_NEXT_PRODUCT_BRANCH}', propagate: false, wait: true, parameters: [
```
Comment (winklerm) [inline job-dsls/jobs/prod/prod_cron_meta_nightly_pipeline.groovy:54]: Should the references to variables be in double quotes or are single quites fine? Sorry, not sure about pipelines..

Comment (Ginxo) [reply job-dsls/jobs/prod/prod_cron_meta_nightly_pipeline.groovy:54]: I know it's hard to get it sometimes @winklerm but the whole jenkins pipeline content is part of the `metaJob` variable which uses triple double quoted `"""` stuff https://groovy-lang.org/syntax.html#_triple_double_quoted_string so variables are going to be replaced during prod_cron_meta_nightly_pipeline.groovy execution.
it is different from `\${env.DEFAULT_CONFIG_BRANCH}` which is going to be executed at runtime.

Comment (winklerm) [reply job-dsls/jobs/prod/prod_cron_meta_nightly_pipeline.groovy:54]: I see, thanks!

**Review by winklerm — APPROVED** [review]

**Pull request merged**